syft/pkg/cataloger/binary/classifier.go

```diff
@@ -91,26 +96,83 @@ func fileNameTemplateVersionMatcher(fileNamePattern string, contentTemplate stri
 			return nil, fmt.Errorf("unable to compile rendered regex=%q: %w", patternBuf.String(), err)
 		}
 
-		contents, err := getContents(reader)
+		contents, err := getContents(resolver, location)
 		if err != nil {
 			return nil, fmt.Errorf("unable to get read contents for file: %w", err)
 		}
 
 		matchMetadata := internal.MatchNamedCaptureGroups(tmplPattern, string(contents))
-		return singlePackage(classifier, reader, matchMetadata), nil
+		return singlePackage(classifier, location, matchMetadata), nil
 	}
 }
 
 func fileContentsVersionMatcher(pattern string) evidenceMatcher {
 	pat := regexp.MustCompile(pattern)
-	return func(classifier classifier, reader source.LocationReadCloser) ([]pkg.Package, error) {
-		contents, err := getContents(reader)
+	return func(resolver source.FileResolver, classifier classifier, location source.Location) ([]pkg.Package, error) {
+		contents, err := getContents(resolver, location)
 		if err != nil {
 			return nil, fmt.Errorf("unable to get read contents for file: %w", err)
 		}
 
 		matchMetadata := internal.MatchNamedCaptureGroups(pat, string(contents))
-		return singlePackage(classifier, reader, matchMetadata), nil
+		return singlePackage(classifier, location, matchMetadata), nil
+	}
+}
+
+func isExecutable(mode os.FileMode) bool {
+	return mode&0111 != 0
```

Comment:
when catalogers look for indications of binaries we've leaned away from using the mode as a way to determine if the file is what we're looking for. Instead MIME type is more preferable of a check here (or no check and let the sharedLibMatcher surface nothing).

Reply:
Right. Unfortunately `internal.IsExecutable` seems not to work, I've just removed this whole check, it was only added because there were a lot of results returned for `**/python*`, but I don't think it will be a huge performance issue, at least it isn't in my testing.